web/app/Models/User.php

```diff
@@ -230,8 +231,12 @@ public function requestContact(User $user_to_request): bool
      */
     public function approveContactRequest(User $user_to_approve): bool
     {
-        Interaction::remove($user_to_approve, InteractionType::USER_CONTACT_REQUESTS, $this);
-        return $this->addContact($user_to_approve);
+        DB::transaction(function () use ($user_to_approve) {
+            Interaction::remove($user_to_approve, InteractionType::USER_CONTACT_REQUESTS, $this);
+            return $this->addContact($user_to_approve);
+        });
```

Comment:
How do returns from `DB::transaction()` work? Why is it just `return false` at the end now?

Reply:
DB::transaction returns whatever return was invoked from the call within, which is already a boolean. The `return false;` at the end is largely to get PHPStorm to shut up because it will either return true or false within, or throw an exception, but PHPStorm complained about no return. I could also add a phpdoc telling it to ignore that inspection, but it's already in a few places here.

Reply:
Interesting, so `DB::transaction();` can act as a `return` in a function, even though it's a call? Or is it because `return` in the lambda returns from the outer function?

Reply:
I should just be returning the result of the call, you're right. https://github.com/laravel/framework/blob/4.1/src/Illuminate/Database/Connection.php#L417 shows how it works, a new commit has it return the result from `DB::transaction()` and it passes all tests.

Reply:
Ah okay, this makes more sense now.